zasady_konstrukcji_api.rst

```diff
@@ -0,0 +1,381 @@
+Zasady konstrukcji API
+======================
+do wewnętrznego użycia w EZD oraz dla zewnętrznych aplikacji — rekomendacje
+>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+
+
+*Dobre praktyki i zalecenia dotyczące konstrukcji API.*
+
+
+API dla Java
+------------
+
+Projektuj przewidując rozwój
+++++++++++++++++++++++++++++
+
+Jeśli API jest cokolwiek warte – będzie ewoluować.   Z drugiej strony, jeśli coś w API już jest, to zapewne będzie musiało zostać.
+
+2 postulaty wstecznej zgodności:
+
+- skompilowany kod nadal działa (zgodność binarna)
+- istniejący kod źródłowy można nadal skompilować (zgodność źródłowa).
+
+Nie da się przewidzieć, w jaki sposób użytkownicy skorzystają z API.
+Rozważając zmianę – należy rozumować konserwatywnie: *tylko* jeśli można stwierdzić z prawdopodobieństwem graniczącym z pewnością, że zmiana nie spowoduje załamania jakiejkolwiek istniejącej aplikacji – *można* zmianę wprowadzić.
+
+Jeśli potrzebne jest wprowadzenie większych zmian – należy zbudować nowe API, z inną nazwą.
+
+Dobrym pomysłem jest wprowadzenie na początek jednej lub kilku wersji o numerach 0.xx przed wersją 1.0. Użytkownicy rozumieją, że takie wersje mogą być zmieniane.
+
+API – cele projektowe
++++++++++++++++++++++
+
+Musi być:
+
+- absolutnie poprawny
+- łatwy do użycia
+- trudny do błędnego użycia
+- łatwy do nauczenia
+- wystarczająco szybki
+- wystarczająco mały.
+
+Minimalizm
+++++++++++
+
+Znacznie łatwiej dodawać nowe elementy niż je usuwać.
+
+Im więcej jest elementów w API, tym trudniej jest się nauczyć.
+
+Im większy API, tym więcej może być błędów.
+
+Nadmiarowe metody niepotrzebnie zajmują miejsce.
+
+Ostrożnie z pakietami
```

Comment:
https://github.com/Ministerstwo-Cyfryzacji/ezd-analizy-it/pull/14/files#r91765472

znów, why Java